spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -75,44 +76,36 @@ protected SQLAdmin sqlAdminService(CredentialsProvider credentialsProvider)
 
 		return new SQLAdmin.Builder(httpTransport, jsonFactory,
 				new HttpCredentialsAdapter(credentialsProvider.getCredentials()))
-				.build();
+						.build();
 	}
 
 	@Bean
 	@ConditionalOnMissingBean(CloudSqlJdbcUrlProvider.class)
 	public CloudSqlJdbcUrlProvider cloudSqlJdbcUrlProvider(SQLAdmin sqlAdmin,
 			GcpProjectIdProvider projectIdProvider)
-			throws IOException, ClassNotFoundException {
+			throws IOException {
 		Assert.hasText(projectIdProvider.getProjectId(),
 				"A project ID must be provided.");
 
-		// TODO(joaomartins): Set the credential factory to be used by SocketFactory when
-		// https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/issues/41 is
-		// resolved. For now, it uses application default creds.
-
-		// Load MySQL driver to register it, so the user doesn't have to.
-		Class.forName("com.mysql.jdbc.Driver");
-
-		// Look for the region if the user didn't specify one.
-		String region = this.gcpCloudSqlProperties.getRegion();
-		if (!StringUtils.hasText(region)) {
-			region = sqlAdmin.instances().get(projectIdProvider.getProjectId(),
-					this.gcpCloudSqlProperties.getInstanceName()).execute().getRegion();
+		String appEngineVersion = System.getProperty("com.google.appengine.runtime.version");
```

Comment:
I actually don't think it should be a separate starter. This is already the "gcp" starter, it makes sense that it's smart enough to distinguish between our different compute environments (but maybe in different AutoConfiguration objects and I mentioned before). For discovering the project ID if the user doesn't specify one, we're already trying out a bunch of different locations depending on the compute environment, by delegating to `ServiceOptions`.

For environments where a lot of things change, like PCF, it could probably make sense to have a new starter, although I'm still not too convinced, since that starter would likely span a bunch of different products (SQL, Pub/Sub, Core, etc.)...